jobbergate_cli/main.py

```diff
@@ -43,8 +74,106 @@ def interactive_get_username_password():
     return username, password
 
 
+def init_cache_dir():
+    """
+    Make sure that the root Jobbergate cache directory exists.
+    """
+    JOBBERGATE_CACHE_DIR.mkdir(exist_ok=True, parents=True)
+
+
+def init_logs(username=None, verbose=False):
+    """
+    Initialize the rotatating file log handler. Logs will be retained for 1 week.
+    """
+    # Remove default stderr handler at level INFO
+    logger.remove()
+    JOBBERGATE_LOG_PATH.parent.mkdir(exist_ok=True)
+
+    if verbose:
+        logger.add(sys.stdout, level="DEBUG")
+
+    logger.add(JOBBERGATE_LOG_PATH, rotation="00:00", retention="1 week", level="DEBUG")
+    logger.debug("Logging initialized")
+    if username:
+        logger.debug(f"  for user {username}")
+
+
+def jobbergate_command_wrapper(func):
+    """Wraps a jobbergate command to include logging, error handling, and user output
+
+    Reports the command being called an its parameters to the user log. Includes log
+    lines about starting and finishing the command. Also reports errors to the user
+    as well as sending the error report to Sentry. Finally, prints any output provided
+    by the called command to stdout.
+    """
+
+    @functools.wraps(func)
+    def wrapper(ctx, *args, **kwargs):
+        try:
+            main_message = f"Handling command '{ctx.command.name}'"
+            if ctx.params:
+                main_message += " with params:"
+            logger.debug(
+                "\n  ".join(
+                    [main_message, *[f"{k}={v}" for (k, v) in ctx.params.items()]]
+                )
+            )
```

Comment:
Good catch

Reply:
Oh, this is all screwed up. will fix

Reply:
No, because we want the main_message printed no matter what. All the `if` logic does is append " with params:" onto th end of main_message. If the params list is empty, the join is essentially a no-op.

Reply:
It is very hard to understand that from the code.

Could you move the `.join` to inside the `if`, so it becomes clear that the params are appended when they exist only, and the message will always be printed?

Reply:
Resolved in next commit. Changed message construction to use a vanilla `for` loop and moved it inside the `if`.